flask_monitoringdashboard/views/reporting.py

```diff
@@ -0,0 +1,104 @@
+from flask_monitoringdashboard.controllers.requests import get_status_code_frequencies_in_interval
+from flask_monitoringdashboard.core.auth import secure
+from datetime import datetime
+from flask import request
+
+from flask_monitoringdashboard import blueprint
+from flask_monitoringdashboard.core.date_interval import DateInterval
+from flask_monitoringdashboard.core.reporting.questions.average_latency import AverageLatency
+
+from flask_monitoringdashboard.core.reporting.questions.status_code_distribution import StatusCodeDistribution
+from flask_monitoringdashboard.database import session_scope
+from flask_monitoringdashboard.database.endpoint import get_endpoints
+
+
+def get_date(p):
+    return datetime.utcfromtimestamp(int(request.args.get(p)))
+
+
+def status_code_report(endpoint, comparison_interval, compared_to_interval):
+    with session_scope() as db_session:
+        comparison_interval_distribution = get_status_code_frequencies_in_interval(db_session, endpoint.id,
+                                                                                   comparison_interval.start_date(),
+                                                                                   comparison_interval.end_date())
+
+        compared_to_interval_distribution = get_status_code_frequencies_in_interval(db_session, endpoint.id,
+                                                                                    compared_to_interval.start_date(),
+                                                                                    compared_to_interval.end_date())
+
+        total_requests_comparison_interval = sum(comparison_interval_distribution.values())
+        total_requests_compared_to_interval = sum(compared_to_interval_distribution.values())
+
+        percentages_comparison_interval = [(status_code, frequency / total_requests_comparison_interval * 100) for
+                                           (status_code, frequency) in
+                                           comparison_interval_distribution.items()]
+
+        percentages_compared_to_interval = [(status_code, frequency / total_requests_compared_to_interval * 100) for
+                                            (status_code, frequency) in
+                                            compared_to_interval_distribution.items()]
+
+        frequencies_comparison_interval = [comparison_interval_distribution[status_code] for status_code in
+                                           sorted(comparison_interval_distribution.keys())]
+
+        frequencies_compared_to_interval = [compared_to_interval_distribution[status_code] for status_code in
+                                            sorted(compared_to_interval_distribution.keys())]
+
+        chisq, p = 0, 0  # chisquare(frequencies_comparison_interval, frequencies_compared_to_interval)
+
+        return dict(
+            significant=float(p) < .05,
+            percentages_comparison_interval=comparison_interval_distribution,
```

Comment:
isn't p always set to zero in the line above?

Reply:
Same comment applies here as in your comment before this one :)